packages/core/src/index.ts

```diff
@@ -690,8 +701,17 @@ class Hopr extends EventEmitter {
     }
 
     const channel = ethereum.getChannel(selfPubKey, counterpartyPubKey)
+    let channelId: Hash
+
+    try {
+      channelId = await channel.open(new Balance(amountToFund))
+    } catch (err) {
+      await this.isOutOfFunds(err)
```

Comment:
Why don't use `isOutOfFundsError()` ?

Reply:
`isOutOfFunds` uses `isOutOfFundsError` but also reacts to the error, `isOutOfFunds` is meant to encapsulate that logic so it's not repeated for every transaction